### ISSUE TITLE
Pagination from page 1 (one-indexed-parameters) not runs well

### ISSUE THREAD

**Issue by efimatica** [spring-projects/spring-data-commons#2889]
Good morning from spain.
My scope is Spring boot 3.1 I'm implemeting REST service and I have pagination for return large list of elements. For default Spring boot manage the page from 0 index. I have reading that is possible start pagination from page 1, using this senetence in application.yaml:
```
spring:
  data:
    web:
      pageable:
        one-indexed-parameters: true
```
With new settings, the page 0 and the page 1 is the same. It show the same result. The problem is that the param "number" on paginations settings always show one page less, that I access:

Examples:

To page 0
```
Request: 
{{base_url}}/system/business/pagination?size=5&page=0

Response details:
    "pagination": {
    "totalPages": 4,
    "totalElements": 17,
    "size": 5,
    "number": 0,
    "numberOfElements": 5,
    "empty": false,
    "first": true,
    "last": false,
    "sorted": false
}

Note: I would prefer that it returns null. 
```

To page 1
```
Request: 
{{base_url}}/system/business/pagination?size=5&page=1

Response details:
    "pagination": {
    "totalPages": 4,
    "totalElements": 17,
    "size": 5,
    "number": 0,
    "numberOfElements": 5,
    "empty": false,
    "first": true,
    "last": false,
    "sorted": false
}

Note: Returns the same that page 0. See that number value is wrong. It would have to be 1. 
```

To page 2
```
Request: 
{{base_url}}/system/business/pagination?size=5&page=2

Response details:
    "pagination": {
    "totalPages": 4,
    "totalElements": 17,
    "size": 5,
    "number": 1,
    "numberOfElements": 5,
    "empty": false,
    "first": false,
    "last": false,
    "sorted": false
}

Note: Return the page 2, but number value is 1. 
```

Thanks for your attention

**Comment by odrotbohm** [comment]
I suspect you serialize a `Page` instance? That is not supported (as discussed in #1683) in the sense that we do not guarantee the serialization format. What you essentially see is the internal structure of the `Page` implementation, un-adapted to the parameter resolution style you have configured. Jackson simply doesn't know about that customization.

To produce proper and stable JSON representations use `PagedResourcesAssembler` (via DI) and call `toModel(Page)`.

**Comment by velykov** [comment]
Same paroblem

**Comment by elijah-pl** [comment]
@odrotbohm

`spring.data.web.pageable.serialization-mode=VIA_DTO` changes nothing. The `number` field is always zero-indexed.

**Comment by fgabolde** [comment]
We are also running into this issue here on Spring Boot 3.5.7 and corresponding spring-data-commons. The official docs mention [manually wrapping](https://docs.spring.io/spring-data/commons/reference/repositories/core-extensions.html#core.web.page) Page instances with PagedModel, which is what we are doing (not `pageSerializationMode = VIA_DTO`). We have the same symptoms:

* binding works well, so page=1 results in a Pageable with pageNumber = 0
* this is nicely translated into our jOOQ query with `.limit(pageable.getPageSize()).offset(pageable.getOffset())`
* but then the PagedModel is serialized with page.number = 0, which is surprising to users

Note we are not importing the whole HATEOAS lib, just spring-data-commons, since we are not using Spring Data for much, hence jOOQ. We just want the nice Pageable binding so all our APIs have the same shape.

I haven't tested page=0 because our OpenAPI spec forbids it, but IIRC before we used PagedModel (we used other tricks to serialize Page instances instead) we already ran into the weirdness that is page=0 being the same as page=1. In fact, I can find in our old code a very custom Jackson converter that just adds 1 to the number field in responses... In the end, we're probably just going to build our own POJO and wrap Page with that instead.

tl;dr: this makes `one-indexed-parameters: true` surprising in its incompleteness, and should be documented near the mentions of PagedModel in the docs linked above.
